projects/packages/connection/src/class-manager.php

```diff
@@ -972,6 +972,20 @@ public function register( $api_endpoint = 'register' ) {
 
 		$this->get_tokens()->update_blog_token( (string) $registration_details->jetpack_secret );
 
+		if ( ! $registration_details->allow_inplace_authorization ) {
+			// Forces register_site REST endpoint to return the Calypso authorization URL.
+			add_filter( 'jetpack_use_iframe_authorization_flow', '__return_false', 20 );
+		}
+
+		add_filter(
+			'jetpack_register_site_rest_response',
+			function ( $response ) use ( $registration_details ) {
+				$response['allowInplaceAuthorization'] = $registration_details->allow_inplace_authorization;
```

Comment:
Not that it's very likely, but what if we change the register response in the future?